service/src/integrationTest/kotlin/fi/espoo/evaka/invoicing/service/InvoiceGeneratorIntegrationTest.kt

```diff
@@ -1497,6 +1536,47 @@ class InvoiceGeneratorIntegrationTest : PureJdbiTest(resetDbBeforeEach = true) {
         }
     }
 
+    @Test
+    fun `absences derived from daily service times with all days off generate a discount on invoices`() {
+        val period = DateRange(LocalDate.of(2019, 1, 1), LocalDate.of(2019, 1, 31))
+
+        initDataForAbsences(listOf(period), listOf(), testChild_1)
+        insertDailyServiceTimes(
+            testChild_1.id,
+            DailyServiceTimes.IrregularTimes(
+                monday = null,
+                tuesday = null,
+                wednesday = null,
+                thursday = null,
+                friday = null,
+                saturday = null,
+                sunday = null,
+                validityPeriod = period
+            )
```

Comment:
I think a test where there is a mix of absences generated by daily service times and manually entered absences would be valuable.

Also, I think this test is just wrong. I don't think these daily service times are really supposed to be interpreted in a way that the child is absent every single weekday.

Reply:
This is not a real world case, no. I added this test as a counter point to the other one, so that we can be sure that something actually happens. 

I'll convert this to mixed absences 👍

Reply:
This test just got me thinking that maybe this exact behaviour is not what we want. The constraint to have at least one of the days filled out was recently taken out so this _could_ happen.

Reply:
It is a real-world scenario where a child has irregular daily service times without any specified times. I am not sure how this should be interpreted, but my best guess is that by default the child is absent on every day, unless a reservation is manually added.

Reply:
jolle's suggestion sounds good to me. Just need to check that a reservation will override the absences derived from daily service times.

Reply:
Tweaked the function to not return absences for days with reservations and added a test for that.

Reply:
I'm sorry about yet another comment, but should we use attendances instead of reservations in this case?